File: template/nestJs/src/application/application.service.ts
import { Application } from '@app/models';
import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { PaginationQueryDto } from './dto/pagination-query.dto';
import { I18nContext, I18nService } from 'nestjs-i18n';
import { I18nTranslations } from '../.generate/i18n.generated';

@Injectable()
export class ApplicationService {
  constructor(
    @InjectRepository(Application)
    private applicationRep: Repository<Application>,
    private readonly i18n: I18nService<I18nTranslations>
  ) {}

  async findAllApplication(searchInfo: PaginationQueryDto) {
    const { page, limit, keywords, classify } = searchInfo;
    const skip = (page - 1) * limit;

    const queryBuilder = this.applicationRep.createQueryBuilder('application');
    if (keywords) {
      const field = ['name', 'description', 'tag'];
      const conditions = field
        .map((item) => `application.${item} LIKE :keywords`)
        .join(' OR ');
      queryBuilder.where(`(${conditions})`, { keywords: `%${keywords}%` });
    }

    if (classify !== 'all') {
      queryBuilder.andWhere('application.classify= :classify', { classify });
    }

    const [data, total] = await queryBuilder
      .skip(skip)
      .take(limit)
      .getManyAndCount();

    return {
      data: data.map((item) => ({
        ...item,
        tag: this.parseTagSafely(item.tag),
      })),
      total,
    };
  }

  async createApplication(dto, isInit: boolean) {
    const { name, description, tag, icon, classify } = dto;
    const applicationInfo = this.applicationRep.findOne({
      where: { name },
    });
    if (isInit === true && (await applicationInfo)) {
      return applicationInfo;
    }
    if ((await applicationInfo) && isInit === false) {
      throw new HttpException(
        `exception.applicationInfo.exists：${name}`,
        HttpStatus.BAD_REQUEST
      );
    }
    return this.applicationRep.save({
      name,
      description,
      tag,
      icon,
      classify,
    });
  }

  private parseTagSafely(tag: string): unknown {
    try {
      return JSON.parse(tag);
    } catch {
      return [];
    }
  }
}

File: template/springboot/src/main/java/com/TinyPro/service/ApplicationService.java
package com.TinyPro.service;

import com.TinyPro.entity.dto.CreateApplicationDto;
import com.TinyPro.entity.dto.PaginationQueryDto;
import com.TinyPro.entity.po.Application;
import com.TinyPro.entity.vo.ApplicationVo;

public interface ApplicationService {
  /**
   * 分页查询应用列表
   */
  ApplicationVo findAllApplication(PaginationQueryDto searchInfo);

  /**
   * 创建应用
   * @param dto 应用信息
   * @param isInit 是否初始化模式（true: 已存在则直接返回，不抛异常；false: 已存在则抛异常）
   * @return 已存在或新建的应用实体
   */
  Application createApplication(CreateApplicationDto dto, boolean isInit);
}
